### PR TITLE
Add org.bionus.Grabber

### PR DESCRIPTION
**Pull request by Bionus**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

Grabber uses `filesystem=host` permission to write downloaded files in paths computed from strings provided by the user (cf [docs](https://www.bionus.org/imgbrd-grabber/docs/filename.html)). I am not familiar with portals.

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.

I am the author and main contributor of the project:
https://github.com/Bionus/imgbrd-grabber/graphs/contributors

- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

Just bought https://www.bionus.org/

- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Bionus** [comment]
bot, build org.bionus.Grabber

**Comment by Bionus** [comment]
The build triggers an `npm install` to embed JS files in the package, which is why I added `--share=network` to the build-args, should I do it another way? Seems like it's not possible in sandbox mode.

**Comment by guihkx** [comment]
Flathub doesn't allow network access during the build step, so all dependencies must be listed beforehand.

You can use [flatpak-builder-tools](https://github.com/flatpak/flatpak-builder-tools) (check the `npm` folder) to generate those npm dependencies into a separate file.

**Comment by barthalion** [comment]
Alternatively, you can extract AppImage. This will make it x86_64-only, you will need to disable aarch64 build as per https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64.

**Comment by Bionus** [comment]
> You can use [flatpak-builder-tools](https://github.com/flatpak/flatpak-builder-tools) (check the node folder) to generate those npm dependencies into a separate file.

Thanks, just did that 👍

> Alternatively, you can extract AppImage.

What's the preferred method if both were to work? Building from source or extracting the AppImage?

**Comment by Bionus** [comment]
bot, build org.bionus.Grabber

**Comment by barthalion** [comment]
> What's the preferred method if both were to work? Building from source or extracting the AppImage?

Some people have strong preference for building from source as it makes the app available for aarch64, but whatever annoys you less is better in my opinion.

File: org.bionus.Grabber.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    XDG_CACHE_HOME: /run/build/grabber/flatpak-node/cache
+    npm_config_cache: /run/build/grabber/flatpak-node/npm-cache
+    npm_config_nodedir: /usr/lib/sdk/node16
+    npm_config_offline: 'true'
+
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (barthalion) [inline org.bionus.Grabber.yml:22]:
```suggestion
  - --socket=fallback-x11
  - --socket=wayland
```

**Pull request closed without merging**